dlt/common/storages/file_storage.py

```diff
@@ -251,3 +254,14 @@ def rmtree_del_ro(action: AnyFun, name: str, exc: Any) -> Any:
                 os.rmdir(name)
             else:
                 os.remove(name)
+
+    @staticmethod
+    def open_zipsafe(path: str, mode: str = "r", encoding: Optional[str] = None, **kwargs: Any) -> IO[Any]:
+        """Opens a file using gzip.open if it is a gzip file, otherwise uses open."""
+        try:
+            f = gzip.open(path, mode, encoding=encoding, **kwargs)
+            # Force gzip to read the first few bytes and check the magic number
+            f.read(2), f.seek(0)
```

Comment:
cool that zipstream supports seek. i wonder how that is implemented (are they using internal buffer, reopen the file etc.)

Reply:
Internal buffer! :)